feat: add plausible analytics support

roedoejet opened the pull request:
Adds plausible support in production. I've also created an account on our Plausible Analytics for this demo page, but it won't show up until gh pages is built with this branch.

roedoejet committed:
feat: add plausible analytics support

joanise commented on projects/word-weaver/src/index.ghpages.html:
```diff
@@ -0,0 +1,176 @@
+
+    <meta name="apple-mobile-web-app-capable" content="yes" />
+
+    <link
```
This code is just craziness. Do we really need this many different sizes of the same image?  I guess this is not a change in this PR, you just duplicated index.html (why?) but bleh.

joanise reviewed (commented):
Given `index.ghpages.html` is an exact copy of `index.html`, why duplicate it? Is this a gh-pages requirement? If it's always going to be the same, I'd copy it as part of the build process, but we didn't have it before so it seems unlikely we really need it.

And yuck, all the images for apple. There must be a simpler way with less content duplication...

roedoejet committed:
feat(analytics): add opt-in and opt-out buttons for analytics

also update our i18n libraries since @biesbjerg is abandoned

fixes https://github.com/roedoejet/kawennonnis_ui/issues/3\#issuecomment-1799639236

this also required that we change from 'marker' to '_' for the new library to notice markers

this commit is missing french translations

joanise commented on projects/word-weaver/src/app/app/app.component.ts:
```diff
@@ -1,5 +1,5 @@
 import { Component, OnInit } from "@angular/core";
-import { marker } from "@biesbjerg/ngx-translate-extract-marker";
+import { marker as _ } from "@colsen1991/ngx-translate-extract-marker";
```
I've seen this syntax recommended somewhere, and I guess we can use it, but I don't like it because it obfuscates stuff. I would prefer continuing to import this as `marker` and spell out marker everywhere we use it.

roedoejet replied:
I agree, but `marker` no longer seemed to be recognized - I had to use `_` by the new `ngx-translate-extract` library

joanise commented on package.json:
```diff
@@ -107,6 +106,7 @@
     "@types/uuid": "^3.4.7",
     "@typescript-eslint/eslint-plugin": "5.43.0",
     "@typescript-eslint/parser": "5.43.0",
+    "@vendure/ngx-translate-extract": "^8.2.3",
```
Makes sense to update to the current maintained version here.

joanise reviewed (commented):
Scanning the code quickly, it looks fine, but I'll have to test it later.

joanise committed:
fix: add French translations for analytics messages

joanise committed:
refactor: use the full "marker" rather than the obscure "_"

joanise reviewed (approved):
Translated now.
I think this will be ready to merge, ideally (but optionally) with #78 too.
